driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedCrudHelper.java

```diff
@@ -0,0 +1,627 @@
+import com.mongodb.client.AggregateIterable;
+import com.mongodb.client.ChangeStreamIterable;
+import com.mongodb.client.ClientSession;
+import com.mongodb.client.FindIterable;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.BulkWriteOptions;
+import com.mongodb.client.model.DeleteManyModel;
+import com.mongodb.client.model.DeleteOneModel;
+import com.mongodb.client.model.DeleteOptions;
+import com.mongodb.client.model.FindOneAndUpdateOptions;
+import com.mongodb.client.model.IndexOptions;
+import com.mongodb.client.model.InsertManyOptions;
+import com.mongodb.client.model.InsertOneModel;
+import com.mongodb.client.model.InsertOneOptions;
+import com.mongodb.client.model.ReplaceOneModel;
+import com.mongodb.client.model.ReplaceOptions;
+import com.mongodb.client.model.ReturnDocument;
+import com.mongodb.client.model.UpdateManyModel;
+import com.mongodb.client.model.UpdateOneModel;
+import com.mongodb.client.model.UpdateOptions;
+import com.mongodb.client.model.WriteModel;
+import com.mongodb.client.result.DeleteResult;
+import com.mongodb.client.result.InsertManyResult;
+import com.mongodb.client.result.InsertOneResult;
+import com.mongodb.client.result.UpdateResult;
+import org.bson.BsonArray;
+import org.bson.BsonDocument;
+import org.bson.BsonElement;
+import org.bson.BsonInt32;
+import org.bson.BsonValue;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+final class UnifiedCrudHelper {
+    private final Entities entities;
+
+    UnifiedCrudHelper(final Entities entities) {
+        this.entities = entities;
+    }
+
+    static ReadConcern asReadConcern(final BsonDocument readConcernDocument) {
+        if (readConcernDocument.size() > 1) {
+            throw new UnsupportedOperationException("Unsupported read concern properties");
+        }
+        return new ReadConcern(ReadConcernLevel.fromString(readConcernDocument.getString("level").getValue()));
+    }
+
+    static WriteConcern asWriteConcern(final BsonDocument writeConcernDocument) {
+        if (writeConcernDocument.size() > 1) {
+            throw new UnsupportedOperationException("Unsupported write concern properties");
+        }
+        return new WriteConcern(writeConcernDocument.getInt32("w").intValue());
+    }
+
+    OperationResult executeListDatabases(final BsonDocument operation) {
+        MongoClient client = entities.getClient(operation.getString("object").getValue());
+
+        if (operation.containsKey("arguments")) {
+            throw new UnsupportedOperationException("Unexpected arguments");
+        }
+
+        try {
+            return OperationResult.of(new BsonArray(client.listDatabases(BsonDocument.class).into(new ArrayList<>())));
```

Comment:
I considered this but while developing the tests it was sometimes useful to set breakpoint in the catch blocks of specific methods, and abstracting the catch would make it more difficult.  Given that I'm inclined to leave it alone.  Let me know what you think.

Reply:
Given this is important due to the chance of false positives and this catch pattern is repeated 21 times - I think its probably worth encapsulating and enforcing the pattern to protect future mistakes.

Debugging could be added to easily enough via intellij and that is a much less common work flow, so I'm not sure its a big impediment. What do you think?

Reply:
I'm convinced.  Done.

Reply:
Changes are larger than initially anticipated, since with lambdas it complains about use of variables that are not effectively final.  So had to move all the required variable initializations out of the switch statements.